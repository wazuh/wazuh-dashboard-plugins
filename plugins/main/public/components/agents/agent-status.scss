.wz-agent-icon-tip {
  margin-left: 5px;
}
