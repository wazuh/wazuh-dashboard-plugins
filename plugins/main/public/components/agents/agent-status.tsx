import React from 'react';
import {
  agentStatusColorByAgentStatus,
  agentStatusLabelByAgentStatus,
} from '../../../common/services/wz_agent_status';
import { ColumnWithStatusIcon } from './column-with-status-icon';
import { EuiIconTip } from '@elastic/eui';
import { AGENT_STATUS_CODE } from '../../../common/constants';
import '../../styles/common.scss';

export const AgentStatus = ({ status, children = null, style = {}, agent }) => {
  const statusCodeAgent = AGENT_STATUS_CODE.find(
    (status: StatusCodeAgent) => status.STATUS_CODE === agent?.status_code,
  );
  return (
    <div
      className='euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow'
      style={style}
    >
      <ColumnWithStatusIcon
        text={children || agentStatusLabelByAgentStatus(status).toLowerCase()}
        color={agentStatusColorByAgentStatus(status)}
        tooltip={agentStatusLabelByAgentStatus(status).toLowerCase()}
      />
      <EuiIconTip
        anchorClassName='wz-margin-left-10'
        aria-label='Description'
        size='m'
        type='questionInCircle'
        color='primary'
        content={statusCodeAgent?.STATUS_DESCRIPTION ?? 'Without information'}
      />
    </div>
  );
};
