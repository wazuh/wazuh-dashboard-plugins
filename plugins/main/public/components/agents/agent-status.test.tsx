import React from 'react';
import { render } from 'enzyme';
import { AgentStatus } from './agent-status';
import {
  UI_COLOR_AGENT_STATUS,
  UI_LABEL_NAME_AGENT_STATUS,
  UI_ORDER_AGENT_STATUS,
} from '../../../common/constants';

describe('AgentStatus component', () => {
  test.each(
    UI_ORDER_AGENT_STATUS.map((status, index) => ({
      status,
      color: UI_COLOR_AGENT_STATUS[status],
      label: UI_LABEL_NAME_AGENT_STATUS[status],
      agent: { status_code: index },
    })),
  )(
    'Renders status indicator with the its color and the label in lower case - %j',
    input => {
      const wrapper = render(
        <AgentStatus status={input.status} agent={input.agent} />,
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('svg').prop('style')).toHaveProperty(
        'color',
        input.color,
      );
      expect(wrapper.find('.hide-agent-status').text()).toEqual(
        input.label.toLowerCase(),
      );
    },
  );
});
