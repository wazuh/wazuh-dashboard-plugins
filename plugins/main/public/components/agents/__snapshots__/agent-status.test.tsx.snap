// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentStatus component Renders status indicator with the its color and the label in lower case - {"status":"active","color":"#007871","label":"Active","agent":{"status_code":0}} 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
>
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--customColor euiIcon-isLoading"
            focusable="false"
            height="16"
            role="img"
            style="color:#007871"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>
  <span
    class="hide-agent-status"
  >
    active
  </span>
  <span
    class="euiToolTipAnchor wz-margin-left-10"
  >
    <svg
      aria-hidden="true"
      aria-label="Description"
      class="euiIcon euiIcon--medium euiIcon--primary euiIcon-isLoading"
      focusable="true"
      height="16"
      role="img"
      tabindex="0"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    />
  </span>
</div>
`;

exports[`AgentStatus component Renders status indicator with the its color and the label in lower case - {"status":"disconnected","color":"#BD271E","label":"Disconnected","agent":{"status_code":1}} 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
>
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--customColor"
            focusable="false"
            height="16"
            role="img"
            style="color:#BD271E"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>
  <span
    class="hide-agent-status"
  >
    disconnected
  </span>
  <span
    class="euiToolTipAnchor wz-margin-left-10"
  >
    <svg
      aria-label="Description"
      class="euiIcon euiIcon--medium euiIcon--primary"
      focusable="true"
      height="16"
      role="img"
      tabindex="0"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 14A6 6 0 1 1 8 2a6 6 0 0 1 0 12Zm0-1A5 5 0 1 0 8 3a5 5 0 0 0 0 10Zm-.186-1.065A.785.785 0 0 1 7 11.12c0-.48.34-.82.814-.82.475 0 .809.34.809.82 0 .475-.334.815-.809.815ZM5.9 6.317C5.96 5.168 6.755 4.4 8.048 4.4c1.218 0 2.091.759 2.091 1.8 0 .736-.36 1.304-1.03 1.707-.56.33-.717.56-.717 1.022v.305l-.1.1H7.47l-.1-.1v-.431c-.005-.646.302-1.104.987-1.514.527-.322.708-.59.708-1.047 0-.536-.416-.91-1.05-.91-.652 0-1.064.374-1.112.998l-.1.092H6l-.1-.105Z"
      />
    </svg>
  </span>
</div>
`;

exports[`AgentStatus component Renders status indicator with the its color and the label in lower case - {"status":"never_connected","color":"#646A77","label":"Never connected","agent":{"status_code":3}} 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
>
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--customColor"
            focusable="false"
            height="16"
            role="img"
            style="color:#646A77"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>
  <span
    class="hide-agent-status"
  >
    never connected
  </span>
  <span
    class="euiToolTipAnchor wz-margin-left-10"
  >
    <svg
      aria-label="Description"
      class="euiIcon euiIcon--medium euiIcon--primary"
      focusable="true"
      height="16"
      role="img"
      tabindex="0"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 14A6 6 0 1 1 8 2a6 6 0 0 1 0 12Zm0-1A5 5 0 1 0 8 3a5 5 0 0 0 0 10Zm-.186-1.065A.785.785 0 0 1 7 11.12c0-.48.34-.82.814-.82.475 0 .809.34.809.82 0 .475-.334.815-.809.815ZM5.9 6.317C5.96 5.168 6.755 4.4 8.048 4.4c1.218 0 2.091.759 2.091 1.8 0 .736-.36 1.304-1.03 1.707-.56.33-.717.56-.717 1.022v.305l-.1.1H7.47l-.1-.1v-.431c-.005-.646.302-1.104.987-1.514.527-.322.708-.59.708-1.047 0-.536-.416-.91-1.05-.91-.652 0-1.064.374-1.112.998l-.1.092H6l-.1-.105Z"
      />
    </svg>
  </span>
</div>
`;

exports[`AgentStatus component Renders status indicator with the its color and the label in lower case - {"status":"pending","color":"#FEC514","label":"Pending","agent":{"status_code":2}} 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
>
  <span
    class="euiToolTipAnchor"
  >
    <div
      class="euiHealth euiHealth--textSizeS wz-flex"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon--customColor"
            focusable="false"
            height="16"
            role="img"
            style="color:#FEC514"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="8"
              r="4"
            />
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
      </div>
    </div>
  </span>
  <span
    class="hide-agent-status"
  >
    pending
  </span>
  <span
    class="euiToolTipAnchor wz-margin-left-10"
  >
    <svg
      aria-label="Description"
      class="euiIcon euiIcon--medium euiIcon--primary"
      focusable="true"
      height="16"
      role="img"
      tabindex="0"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 14A6 6 0 1 1 8 2a6 6 0 0 1 0 12Zm0-1A5 5 0 1 0 8 3a5 5 0 0 0 0 10Zm-.186-1.065A.785.785 0 0 1 7 11.12c0-.48.34-.82.814-.82.475 0 .809.34.809.82 0 .475-.334.815-.809.815ZM5.9 6.317C5.96 5.168 6.755 4.4 8.048 4.4c1.218 0 2.091.759 2.091 1.8 0 .736-.36 1.304-1.03 1.707-.56.33-.717.56-.717 1.022v.305l-.1.1H7.47l-.1-.1v-.431c-.005-.646.302-1.104.987-1.514.527-.322.708-.59.708-1.047 0-.536-.416-.91-1.05-.91-.652 0-1.064.374-1.112.998l-.1.092H6l-.1-.105Z"
      />
    </svg>
  </span>
</div>
`;
